File: HeartBeats/src/app/Models/Donor.ts
export interface Donor {
  
  id: number;
  name?: string | null;
  number?: string | null;
  email?: string | null;
  dob: Date;
  blood?: string | null;
  gender?: string | null;
  state?: string | null;
  city?: string | null;
  alcoholic?: string | null;
  password?: string | null;
  // You can add other properties as needed

  // Flags for Blood Donation Request and Report
  bloodDonationRequested: boolean;
  reportReason?: string;
  }
  

File: HeartBeats API/HeartBeatsv2/Models/Donor.cs
using Microsoft.EntityFrameworkCore.Metadata.Internal;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace HeartBeatsv2.Models;

public partial class Donor
{
    public int Donorid { get; set; }

    public string Name { get; set; } = null!;

    public string Number { get; set; } = null!;

    public string Email { get; set; } = null!;

    public DateTime Dob { get; set; }

    public string Blood { get; set; } = null!;

    public string Gender { get; set; } = null!;

    public string City { get; set; } = null!;

    public string State { get; set; } = null!;

    public string Alcoholic { get; set; } = null!;

    public string Password { get; set; } = null!;

    [System.Text.Json.Serialization.JsonIgnore]
    public virtual ICollection<Report> Reports { get; set; } = new List<Report>();

    [System.Text.Json.Serialization.JsonIgnore]
    public virtual ICollection<Request> Requests { get; set; } = new List<Request>();
}
